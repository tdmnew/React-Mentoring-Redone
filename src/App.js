import "./App.scss";
import React from "react";

import Header from "./Components/Header/Header.js";
import Nav from "./Components/Nav/Nav.js";
import MovieList from "./Components/MovieList/MovieList.js";
import Footer from "./Components/Footer/Footer.js";

import ErrorBoundary from "./Components/ErrorBoundary.js";

import mockMovies from "./movies.json";

//Modals
import Modal from "./Components/Modals/RootModal";
import { ModalContext } from "./Context/ModalContext.js";

export default function App() {
    const [movies, setMovies] = React.useState(mockMovies);
    
    React.useEffect(() => {
        setMovies([...movies.sort((a, b) => (a.releasedate > b.releasedate) ? 1 : -1)])
    },[])

    function sort(e) {
        e.preventDefault();
        const sortBy = e.target.value;

        switch(sortBy) {
            case 'TITLE':
                setMovies([...movies.sort((a, b) => (a.title > b.title) ? 1 : -1)])
                break;
            case 'GENRE':
                setMovies([...movies.sort((a, b) => (a.genre > b.genre) ? 1 : -1)])
                break;
            case 'RELEASE DATE':
                setMovies([...movies.sort((a, b) => (a.releasedate > b.releasedate) ? 1 : -1)])
                break;
            default:
                throw new Error('No case found');
        }
    }

    return (
        <>
            <ModalContext>
                <Header />
                <Nav sort={sort} />
                <ErrorBoundary>
                    <MovieList movies={movies} />
                    <Modal movies={movies} setMovies={setMovies} />
                </ErrorBoundary>
                <Footer />
            </ModalContext>
        </>
    );
}

