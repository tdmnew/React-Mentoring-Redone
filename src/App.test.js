import React from 'react';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import App from './App';
import store from './Store/index.js';

it('Renders matching snapshot', () => {
    const tree = render(
        <Provider store={store}>
            <App />
        </Provider>
    );
    expect(tree).toMatchSnapshot();
});
