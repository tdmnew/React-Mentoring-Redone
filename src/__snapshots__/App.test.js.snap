// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders matching snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <div
          class="header__top"
        >
          <h1
            class="header__top logo"
          >
            <span
              class="header__top logo--left"
            >
              netflix
            </span>
            <span
              class="header__top logo--right"
            >
              roulette
            </span>
          </h1>
          <button
            class="header__top btn"
          >
            + ADD MOVIE
          </button>
        </div>
        <div
          class="search"
        >
          <h2
            class="search title"
          >
            FIND YOUR MOVIE
          </h2>
          <div
            class="search search-bar"
          >
            <input
              class="search search-bar__input"
              placeholder="What do you want to watch?"
              value=""
            />
            <button
              class="search search-bar__btn"
            >
              SEARCH
            </button>
          </div>
        </div>
      </div>
      <div
        class="nav"
      >
        <ul
          class="nav__links"
        >
          <li>
            <a
              href="#All"
            >
              ALL
            </a>
          </li>
        </ul>
        <div
          class="dropdown"
        >
          <span
            class="dropdown__heading"
          >
            SORT BY
          </span>
          <select
            class="dropdown__options"
          >
            <option
              value="release_date"
            >
              RELEASE DATE
            </option>
            <option
              value="title"
            >
              TITLE
            </option>
            <option
              value="genres"
            >
              GENRE
            </option>
            <option
              value="vote_average"
            >
              RATING
            </option>
          </select>
        </div>
      </div>
      <div
        class="no-movies"
      >
        <span
          class="no-movies__text"
        >
          No Movie Found
        </span>
      </div>
      <div
        class="footer"
      >
        <h4
          class="footer__logo"
        >
          <span
            class="footer__logo--left"
          >
            netflix
          </span>
          <span
            class="footer__logo--right"
          >
            roulette
          </span>
        </h4>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <div
        class="header__top"
      >
        <h1
          class="header__top logo"
        >
          <span
            class="header__top logo--left"
          >
            netflix
          </span>
          <span
            class="header__top logo--right"
          >
            roulette
          </span>
        </h1>
        <button
          class="header__top btn"
        >
          + ADD MOVIE
        </button>
      </div>
      <div
        class="search"
      >
        <h2
          class="search title"
        >
          FIND YOUR MOVIE
        </h2>
        <div
          class="search search-bar"
        >
          <input
            class="search search-bar__input"
            placeholder="What do you want to watch?"
            value=""
          />
          <button
            class="search search-bar__btn"
          >
            SEARCH
          </button>
        </div>
      </div>
    </div>
    <div
      class="nav"
    >
      <ul
        class="nav__links"
      >
        <li>
          <a
            href="#All"
          >
            ALL
          </a>
        </li>
      </ul>
      <div
        class="dropdown"
      >
        <span
          class="dropdown__heading"
        >
          SORT BY
        </span>
        <select
          class="dropdown__options"
        >
          <option
            value="release_date"
          >
            RELEASE DATE
          </option>
          <option
            value="title"
          >
            TITLE
          </option>
          <option
            value="genres"
          >
            GENRE
          </option>
          <option
            value="vote_average"
          >
            RATING
          </option>
        </select>
      </div>
    </div>
    <div
      class="no-movies"
    >
      <span
        class="no-movies__text"
      >
        No Movie Found
      </span>
    </div>
    <div
      class="footer"
    >
      <h4
        class="footer__logo"
      >
        <span
          class="footer__logo--left"
        >
          netflix
        </span>
        <span
          class="footer__logo--right"
        >
          roulette
        </span>
      </h4>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
